### PR TITLE
Update amp

### PR DESCRIPTION
**Pull request by joanhey**
Update: 
* PHP v7.3.
* Amp LIbs
* Libuv instead of libev
* the code

Ping @kelunik to review

### CODE REVIEW

File: frameworks/PHP/amp/server.php [hunk 1]
```diff
@@ -88,15 +95,15 @@ private function doHandleRequest($request) {
 
             $items = [];
 
-            $statement = yield $this->mysql->prepare("SELECT * FROM World WHERE id = ?");
+            $statement = yield $this->mysql->prepare('SELECT * FROM World WHERE id = ?');
 
-            for ($i = 0; $i < $queries; $i++) {
```
[the remark this answers is not in the document]
Comment (joanhey) [reply frameworks/PHP/amp/server.php]: We don't use the `$i` variable.
The `while($queries--)` is faster.

File: frameworks/PHP/amp/server.php [hunk 2]
```diff
@@ -12,42 +12,46 @@
...
-use Psr\Log\NullLogger;
+use Monolog\Logger;
+
+define('DB_HOST', gethostbyname('tfb-database'));
 
 Amp\Loop::run(function () {
     $sockets = yield [
-        Cluster::listen("0.0.0.0:8080"),
-        Cluster::listen("[::]:8080"),
+        Cluster::listen('0.0.0.0:8080')
```
Comment (kelunik) [inline frameworks/PHP/amp/server.php]: This should probably pass a customized `BindContext` and increase the default backlog size.

https://github.com/amphp/socket/blob/abc18f1786363ceee2555ffd7fe87e4eead2ce24/src/BindContext.php#L12

Comment (joanhey) [reply frameworks/PHP/amp/server.php]: Yes, 128 is very low for a server.
But I don't know enough Amp.
Please could you help with that?

Comment (kelunik) [reply frameworks/PHP/amp/server.php]: Sure. All you need to do is create an instance of `(new Amp\Socket\BindContext)->withBacklog($size)` as second  parameter.

Comment (joanhey) [reply frameworks/PHP/amp/server.php]: Added socket bind context with `backlog(102400)` as the stream server use 2 sockets.
And with `reuseport` and `tcpnodelay`.

But I miss and option to use `keepalive` in the `BindContest class`.

Comment (kelunik) [reply frameworks/PHP/amp/server.php]: @joanhey I stumbled upon that yesterday, too, so I just opened https://github.com/amphp/socket/issues/67.

Comment (joanhey) [reply frameworks/PHP/amp/server.php]: Without `keepalive`, never will be fast, as each request need to open a new tcp connection.

Comment (kelunik) [reply frameworks/PHP/amp/server.php]: Oh, we're talking about different keep alives. HTTP/1.1 keep alive / connection reuse is automatically enabled.

The thing I just opened is about Tcp keep alive and only affects idle connections.

**Pull request merged**